src/app/frontend/common/components/middleellipsis/middleellipsis.js

```diff
@@ -0,0 +1,71 @@
+// Copyright 2015 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @param {!Element} container
+ * @param {!Element} element
+ * @param {!Function} filter - middle ellipsis filter function used to truncate string
+ *
+ * @return {number}
+ */
+export default function computeTextLength(container, element, filter) {
+  let availableWidth = container.offsetWidth;
+  // Browsers are using floating numbers and element returns integer.
+  // Reduce by 1px in case of rounding problem.
+  let width = element.offsetWidth - 1;
+
+  // If it already fits then do not change
+  if (availableWidth >= width) {
+    return element.textContent.trim().length;
+  }
+
+  return binarySearchLength(availableWidth, element, filter);
+}
+
+/**
+ * Does binary search to find minimal integer I such that given string with length I fits into
+ * available space and with length I + 1 it does not.
+ *
+ * @param {number} availableWidth
+ * @param {!Element} element
+ * @param {!Function} filter
+ *
+ * @return {number}
+ */
+function binarySearchLength(availableWidth, element, filter) {
```

Comment:
I'm trying believe me :) For now i'm struggling with jasmine-jquery in order to test this function because it manipulates DOM directly... This is not fun :)